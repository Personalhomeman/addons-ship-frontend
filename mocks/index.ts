import { AppVersion } from '@/models';
import { TestDevice } from '@/models/test-device';

export const mockAppVersion: AppVersion = {
  id: 123,
  version: '1.0.3',
  platform: 'ios',
  buildNumber: 32,
  buildSlug: '13245ads',
  lastUpdate: new Date('2018-09-22T12:42:31Z'),
  description: 'Some semi-long description',
  whatsNew: 'Some longer whats-new',
  minimumOs: '10.2',
  minimumSdk: 'dunno',
  packageName: 'com.package.name',
  certificateExpires: new Date('2018-09-22T12:42:31Z'),
  bundleId: 'bundle-id',
  size: 4567034,
  supportedDeviceTypes: ['iphone', 'ipad'],
  distributionType: 'development',
  iconUrl: 'http://placekitten.com/160/160',
  appName: 'Standup Timer',
  promotionalText: 'Promotional Text',
  keywords: 'Keywords',
  reviewNotes: 'Review Notes',
  supportUrl: 'https://bitrise.io/support',
  marketingUrl: 'https://bitrise.io/marketing',
  scheme: 'piramid',
  configuration: 'canary',
  publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
};

export const mockTestDevices: TestDevice[] = [
  { deviceId: 'some-device-id-01', deviceType: 'ios', owner: 'test-user-1' },
  { deviceId: 'some-device-id-02', deviceType: 'android', owner: 'test-user-1' },
  { deviceId: 'some-device-id-03', deviceType: 'ios', owner: 'test-user-2' }
];
