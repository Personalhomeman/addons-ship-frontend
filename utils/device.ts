import * as ReactDeviceDetect from 'react-device-detect';
import * as CompareVersions from 'compare-versions';

export const isAndroid = () => ReactDeviceDetect.isAndroid;
export const isIOS = () => ReactDeviceDetect.isIOS;
export const osVersion = () => ReactDeviceDetect.osVersion;
export const mobileModel = () => ReactDeviceDetect.mobileModel;
export const compareVersions = (firstVersion: string, secondVersion: string) => CompareVersions(firstVersion, secondVersion)
