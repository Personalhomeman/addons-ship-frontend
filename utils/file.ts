import fetch from 'node-fetch';

export const uploadFileToS3 = async (file: File | undefined, presignedUrl: string | undefined) => {
  if (!file || !presignedUrl) throw new Error('File or presignedUrl is invalid');

  const options = {
    method: 'put',
    body: file as any
  };

  return fetch(presignedUrl, options).then(resp => {
    if (!resp.ok) throw new Error(`${resp.status}: ${resp.statusText}`);
  });
};
