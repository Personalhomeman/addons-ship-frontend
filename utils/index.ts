import mapKeys from 'lodash/mapKeys';
import mapValues from 'lodash/mapValues';
import camelCase from 'lodash/camelCase';
import snakeCase from 'lodash/snakeCase';
import isPlainObject from 'lodash/isPlainObject';

export const actionTypeCreator = (prefix: string) => (actionType: TemplateStringsArray) => `${prefix}_${actionType}`;

export const camelizeKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => camelCase(key)) as unknown) as T;

export const camelizeKeysDeep = <T>(input: Object): T =>
  mapValues(camelizeKeys(input), (value: any) => (isPlainObject(value) ? camelizeKeysDeep(value) : value)) as T;

export const snakifyKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => snakeCase(key)) as unknown) as T;

export const snakifyKeysDeep = <T>(input: Object): T =>
  mapValues(snakifyKeys(input), (value: any) => (isPlainObject(value) ? snakifyKeysDeep(value) : value)) as T;
