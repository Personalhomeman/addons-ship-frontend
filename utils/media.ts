import { useState, useEffect } from 'react';

type Handler = (e: MediaQueryListEvent) => void;

const getListeners = (queries: string[], handler: Handler) =>
  queries.map(query => {
    const listener = window.matchMedia(query);

    listener.addListener(handler);

    return listener;
  });

export const mediaQuery = (...rules: string[]) => {
  const mediaRules = rules.map(r => `(min-width: ${r})`);

  const stateHandlers = rules.map(() => useState(false));

  const getStateHandler = (media: string) => {
    const idx = mediaRules.indexOf(media);

    return stateHandlers[idx];
  };

  useEffect(() => {
    const handler = ({ matches, media }: MediaQueryListEvent) => {
      const [, set] = getStateHandler(media);

      set(matches);
    };

    const listeners = getListeners(mediaRules, handler);

    listeners.forEach(({ media, matches }) => {
      const [, set] = getStateHandler(media);
      set(matches);
    });

    return () => listeners.forEach(l => l.removeListener(handler));
  });

  return stateHandlers.map(([isMatch]) => isMatch);
};
