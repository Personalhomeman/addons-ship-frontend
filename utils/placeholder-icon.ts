import { placeholderIcons } from '@/config';
import { ProjectType } from '@/models/app';

export const getIcon = (appIcon?: string, projectType?: ProjectType) => {
  let icon = appIcon;

  if (!icon) {
    icon = placeholderIcons[projectType!] || placeholderIcons.other;
  }

  const suffix = Math.random()
    .toString(16)
    .replace(/^0\./, '');

  return `${icon}?${suffix}`;
};
