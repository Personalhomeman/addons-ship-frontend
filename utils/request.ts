import fetch, { RequestInit } from 'node-fetch';

const getHeaders = (token: string) => ({
  Authorization: token
});

interface RequestParams extends RequestInit {
  url: string;
  token: string;
}

export const request = ({ url, method, token, body }: RequestParams) =>
  fetch(url, {
    method,
    headers: getHeaders(token),
    body
  });

export const get = (url: string, token: string) => request({ method: 'get', url, token });
export const del = (url: string, token: string) => request({ method: 'delete', url, token });
export const patch = (url: string, token: string) => request({ method: 'patch', url, token });
export const post = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'post', url, token, body });
export const put = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'put', url, token, body });
