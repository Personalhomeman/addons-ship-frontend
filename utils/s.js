// ðŸ”‡ http://bit.ly/2Q5Gfuh

// prettier-ignore
export default (m) => (Math.random() > 0.01 ? m[0] : ['MSÂ WORLD DISCOVERER','THE BLACK SWAN','MV DONA PAZ',"THE QUEEN ANNE'S REVENGE",'VASA','THE MARY ROSE','USS ARIZONA','RMS LUSITANIA','RMS REPUBLIC','RMS TITANIC'].sort(() => Math.random() - .5).shift());
