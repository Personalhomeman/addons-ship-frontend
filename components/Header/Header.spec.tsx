jest.mock('@/utils/media');
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(input => [input, jest.fn()]),
  useEffect: jest.fn((fn: Function) => fn())
}));
jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  events: {
    on: jest.fn(),
    off: jest.fn()
  },
  useRouter: jest.fn(() => ({ route: '' }))
}));
jest.mock(
  'popper.js',
  () =>
    class {
      constructor() {
        return {
          scheduleUpdate: jest.fn(),
          update: jest.fn()
        };
      }
    }
);

import React, { useState } from 'react';
import Router, { useRouter } from 'next/router';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { AddonBeam, Link } from '@bitrise/bitkit';

import { mediaQuery } from '@/utils/media';
import { mockApp, mockAppVersion } from '@/mocks';

import { Header, Props } from '.';

describe('Header', () => {
  const defaultProps: Props = { app: mockApp, appVersion: mockAppVersion, shouldShowSettingsOnboarding: false };

  beforeEach(() => {
    (mediaQuery as jest.Mock).mockReturnValue([true]);
  });

  it('renders correctly', () => {
    (mediaQuery as jest.Mock).mockReturnValueOnce([true]);

    const tree = shallowToJson(shallow(<Header {...defaultProps}>My app</Header>));
    expect(tree).toMatchSnapshot();
  });

  describe('on mobile', () => {
    beforeEach(() => {
      (mediaQuery as jest.Mock).mockReturnValueOnce([false]);
    });

    it('renders correctly on mobile', () => {
      const tree = shallowToJson(shallow(<Header {...defaultProps}>My app</Header>));
      expect(tree).toMatchSnapshot();
    });

    it('opens the hamburger menu', () => {
      const setHamburgerIconActive = jest.fn();
      (useState as jest.Mock).mockReturnValueOnce([false, setHamburgerIconActive]);

      const wrapper = shallow(<Header {...defaultProps}>My app</Header>);
      (wrapper.find(AddonBeam).props() as any).onHamburgerIconClick();

      expect(setHamburgerIconActive).toHaveBeenCalledWith(true);
    });
  });

  describe('routes', () => {
    const routes = ['/settings', '/version', '/app'];

    routes.forEach(route => {
      test(`${route} for desktop`, () => {
        (useRouter as jest.Mock).mockReturnValueOnce({ route });
        (mediaQuery as jest.Mock).mockReturnValueOnce([true]);
        const tree = shallowToJson(shallow(<Header {...defaultProps} />));
        expect(tree).toMatchSnapshot();
      });

      test(`${route} for mobile`, () => {
        (useRouter as jest.Mock).mockReturnValueOnce({ route });
        (mediaQuery as jest.Mock).mockReturnValueOnce([false]);
        const tree = shallowToJson(shallow(<Header {...defaultProps} />));
        expect(tree).toMatchSnapshot();
      });
    });
  });

  test('handleRouteChange', async () => {
    const setHamburgerIconActive = jest.fn();
    (useState as jest.Mock).mockReturnValueOnce([true, setHamburgerIconActive]);
    let handler;
    (Router.events.on as jest.Mock).mockImplementationOnce((event, _handler) => {
      console.log('Router.events.on', { event, _handler });
      handler = _handler;
    });

    await shallow(<Header {...defaultProps} />);
    expect(typeof handler).toBe('function');

    // @ts-ignore
    await handler();

    expect(setHamburgerIconActive).toHaveBeenCalledWith(false);
  });

  describe('when settings onboarding should be shown', () => {
    beforeEach(() => {
      (mediaQuery as jest.Mock).mockReturnValue([true]);
    });

    it('shows the notification', () => {
      const tree = shallowToJson(shallow(<Header {...defaultProps} shouldShowSettingsOnboarding />));
      expect(tree).toMatchSnapshot();
    });

    it('hides notification on close button selection', async () => {
      // Needed for the tooltip
      (global as any).document.createRange = () => ({
        setStart: () => {},
        setEnd: () => {},
        commonAncestorContainer: {
          nodeName: 'BODY',
          ownerDocument: document
        }
      });

      const setSettingsOnboardingNotificationVisible = jest.fn();
      (useState as jest.Mock).mockReturnValue([true, setSettingsOnboardingNotificationVisible]);

      const tree = await mount(<Header {...defaultProps} shouldShowSettingsOnboarding />);
      (tree.find(Link).props() as any).onClick();

      expect(setSettingsOnboardingNotificationVisible).toHaveBeenCalledWith(false);
    });
  });
});
