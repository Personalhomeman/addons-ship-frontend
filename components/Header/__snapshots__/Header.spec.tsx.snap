// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header on mobile renders correctly on mobile 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={true}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x8"
  >
    <_default
      projectType="other"
      smaller={false}
      title="Awesome app"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;

exports[`Header renders correctly 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={false}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x8"
  >
    <_default
      projectType="other"
      smaller={false}
      title="Awesome app"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;

exports[`Header routes /app for desktop 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={false}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x8"
  >
    <_default
      projectType="other"
      smaller={false}
      title="Awesome app"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;

exports[`Header routes /app for mobile 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={true}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x8"
  >
    <_default
      projectType="other"
      smaller={false}
      title="Awesome app"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;

exports[`Header routes /settings for desktop 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={false}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x5"
  >
    <Flex
      direction="horizontal"
      paddingHorizontal="x0"
    >
      <Link
        as="/apps/test-app-slug-1"
        href="/app?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            className="breadcrumbs"
            size="x4"
          >
            « 
            Awesome app
          </Text>
        </a>
      </Link>
    </Flex>
    <_default
      projectType="settings"
      smaller={true}
      title="Settings"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;

exports[`Header routes /settings for mobile 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={true}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x5"
  >
    <Flex
      direction="horizontal"
      paddingHorizontal="x4"
    >
      <Link
        as="/apps/test-app-slug-1"
        href="/app?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            className="breadcrumbs"
            size="x3"
          >
            « 
            Awesome app
          </Text>
        </a>
      </Link>
    </Flex>
    <_default
      projectType="settings"
      smaller={true}
      title="Settings"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;

exports[`Header routes /version for desktop 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={false}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x5"
  >
    <Flex
      direction="horizontal"
      paddingHorizontal="x0"
    >
      <Link
        as="/apps/test-app-slug-1"
        href="/app?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            className="breadcrumbs"
            size="x4"
          >
            « 
            Awesome app
          </Text>
        </a>
      </Link>
    </Flex>
    <_default
      projectType="other"
      smaller={true}
      title="Awesome app v1.0.3 (32)"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;

exports[`Header routes /version for mobile 1`] = `
<Base>
  <AddonBeam
    addonIcon="SecurityShield"
    addonName="Ship"
    appImage="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
    appLink="https://app.bitrise.io/app/test-app-slug-1"
    appName="Awesome app"
    backgroundColor="grape-4"
    color="white"
    isHamburgerIconActive={false}
    isInResponsiveView={true}
    onHamburgerIconClick={[Function]}
  >
    <Link
      as="/apps/test-app-slug-1/settings"
      href="/settings?appSlug=test-app-slug-1"
    >
      <a>
        <AddonBeamLink
          Component="div"
          icon="Settings"
        >
          Settings
        </AddonBeamLink>
      </a>
    </Link>
  </AddonBeam>
  <Flex
    className="header"
    direction="vertical"
    gap="x2"
    paddingVertical="x5"
  >
    <Flex
      direction="horizontal"
      paddingHorizontal="x4"
    >
      <Link
        as="/apps/test-app-slug-1"
        href="/app?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            className="breadcrumbs"
            size="x3"
          >
            « 
            Awesome app
          </Text>
        </a>
      </Link>
    </Flex>
    <_default
      projectType="other"
      smaller={true}
      title="Awesome app v1.0.3 (32)"
    />
  </Flex>
  <Flex
    alignChildrenVertical="end"
    backgroundColor="grape-5"
    className="mobileMenu"
    color="white"
    direction="vertical"
  >
    <Flex
      padding="x4"
    >
      <a
        href="https://app.bitrise.io/app/test-app-slug-1"
        target="_blank"
      >
        <Flex
          alignChildrenVertical="middle"
          direction="horizontal"
          gap="x2"
        >
          <Image
            borderRadius="x1"
            height="1.5rem"
            src="https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png"
            width="1.5rem"
          />
          <Flex
            alignChildrenVertical="middle"
            direction="horizontal"
            gap="x1"
          >
            <Text
              size="x3"
            >
              Awesome app
            </Text>
            <Icon
              name="OpenInBrowser"
              size="1.5rem"
            />
          </Flex>
        </Flex>
      </a>
    </Flex>
    <Divider
      color="grape-4"
      direction="horizontal"
      width="0.125rem"
    />
    <Flex
      padding="x4"
    >
      <Link
        as="/apps/test-app-slug-1/settings"
        href="/settings?appSlug=test-app-slug-1"
      >
        <a>
          <Text
            color="gray-1"
            size="x3"
          >
            Settings
          </Text>
        </a>
      </Link>
    </Flex>
  </Flex>
</Base>
`;
